core/shared/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -137,6 +142,14 @@ object StaticFile {
     })
   }
 
+  private def calcETagURL[F[_]](implicit F: Sync[F]): URL => F[ETag] = url => {
+    val urlConn = url.openConnection
```

Comment:
This method can throw an `IOException`. I wonder if it should be wrapped in `F.blocking` too.

https://docs.oracle.com/javase/8/docs/api/java/net/URL.html#openConnection--

Reply:
@armanbilge good call 👍🏻

Reply:
Fixed.